src/Compilers/CSharp/Portable/Operations/CSharpOperationNodes.cs

```diff
@@ -1482,6 +1483,41 @@ protected override IOperation CreateValue()
         }
     }
 
+    /// <summary>
+    /// Represents a C# recursive pattern.
+    /// </summary>
+    internal sealed partial class CSharpLazyRecursivePatternOperation : LazyRecursivePatternOperation
+    {
+        private readonly CSharpOperationFactory _operationFactory;
+        private readonly ImmutableArray<BoundSubpattern> _deconstructionSubpatterns;
+        private readonly ImmutableArray<BoundSubpattern> _propertySubpatterns;
+        public CSharpLazyRecursivePatternOperation(
+            CSharpOperationFactory operationFactory,
+            ITypeSymbol inputType,
+            ITypeSymbol matchedType,
+            ISymbol deconstructSymbol,
+            ImmutableArray<BoundSubpattern> deconstructionSubpatterns,
+            ImmutableArray<BoundSubpattern> propertySubpatterns,
+            ISymbol declaredSymbol,
+            SemanticModel semanticModel,
+            SyntaxNode syntax,
+            bool isImplicit)
+            : base(inputType, matchedType, deconstructSymbol, declaredSymbol, semanticModel, syntax, isImplicit)
+        {
+            _operationFactory = operationFactory;
+            _deconstructionSubpatterns = deconstructionSubpatterns;
+            _propertySubpatterns = propertySubpatterns;
+        }
+        public override ImmutableArray<IPatternOperation> CreateDeconstructionSubpatterns()
+        {
+            return _deconstructionSubpatterns.SelectAsArray(p => (IPatternOperation)_operationFactory.Create(p.Pattern));
```

Comment:
>p [](start = 60, length = 1)

nit: `p` should be `d`

Reply:
No, it is a subpattern.

---
In reply to: [244034917](https://github.com/dotnet/roslyn/pull/31967#discussion_r244034917) [](ancestors = 244034917)